### PR TITLE
Add `implicitly_unwrapped_optional` rule

### PR DESCRIPTION
**Pull request by SiarheiFedartsou**
Add `implicitly_unwrapped_optional` rule that warns when using implicitly unwrapped optional,
except cases when this IUO is IBOutlet. 
Also some small fixes to conform to new rule(NSRegularExpression! substituted with NSRegularExpression?).

### CODE REVIEW

**Comment by SiarheiFedartsou** [comment]
It seems that tests failed not because of my changes...

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,11 @@
 
 ##### Enhancements
 
+* Add `implicitly_unwrapped_optional` rule
+  that warns when using implicitly unwrapped optional,
+  except cases when this IUO is IBOutlet.
```
Comment (marcelofabri) [inline CHANGELOG.md]: requires two trailing spaces as described in CONTRIBUTING.md: https://github.com/realm/SwiftLint/blob/master/CONTRIBUTING.md#tracking-changes

File: Source/SwiftLintFramework/Rules/ImplicitlyUnwrappedOptionalRule.swift [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    public func validate(file: File, kind: SwiftDeclarationKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        guard SwiftDeclarationKind.variableKinds().contains(kind) else {
+            return []
+        }
+
+        guard let typeName = dictionary.typeName  else { return [] }
+        guard typeName.hasSuffix("!") else { return [] }
+
+        let isOutlet = dictionary.enclosedSwiftAttributes.contains("source.decl.attribute.iboutlet")
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/ImplicitlyUnwrappedOptionalRule.swift]: what do you think about making this configurable? I think by default we should skip outlets, but I know some people who think that IUO shouldn't be used even on outlets.

**Comment by SiarheiFedartsou** [comment]
@marcelofabri Thanks for your feedback! Just updated pull request.

**Comment by SiarheiFedartsou** [comment]
@marcelofabri Any updates?

**Comment by SiarheiFedartsou** [comment]
@marcelofabri  just updated the PR.

**Comment by marcelofabri** [comment]
Thanks for implementing this! 👏

**Comment by jpsim** [comment]
Thanks for the PR @SiarheiFedartsou!

---

> Can you also revert the changes that were made in the project to make it conform to the new rule?
> 
> I don't think we want to enforce it here (at least for now), as there're only a few places that use IUO and I'm not sure the proposed approaches are really better.

👍  thanks for making this point @marcelofabri!

**Pull request merged**